src/wp-includes/html-api/class-wp-html-tag-processor.php

```diff
@@ -1725,6 +1739,32 @@ private function parse_next_tag() {
 					return true;
 				}
 
+				if (
+					$this->is_inside_foreign_content &&
+					strlen( $html ) > $at + 8 &&
+					'[' === $html[ $at + 2 ] &&
+					'C' === $html[ $at + 3 ] &&
+					'D' === $html[ $at + 4 ] &&
+					'A' === $html[ $at + 5 ] &&
+					'T' === $html[ $at + 6 ] &&
+					'A' === $html[ $at + 7 ] &&
+					'[' === $html[ $at + 8 ]
+				) {
+					$closer_at = strpos( $html, ']]>', $at + 1 );
```

Comment:
Minor, but this could start checking after we find the cdata opener:

```suggestion
					$closer_at = strpos( $html, ']]>', $at + 9 );
```